dsym: init at 1.0.3

nerdbude opened the pull request:
## Motivation for this change

This PR adds `dsym`, a tool to manage dotfiles with Git, to nixpkgs.  
It simplifies managing dotfiles by providing a clean CLI interface.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [X] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.


---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Commit:
dsym: init at 1.0.3

acid-bong reviewed (changes requested):
Hi there, hello. Bunch of notes here.

Also, new modules must be mentioned in the upcoming release notes

acid-bong commented:
Also answer (at least to yourself), is it a good idea to [add alpha-quality software to Nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#quick-start-to-adding-a-package) (first item of the first list) yet, especially with a module?

Commit:
Add 0x17 to the maintainer list

fooker commented on pkgs/by-name/ds/dsym/default.nix:
```diff
@@ -0,0 +1,35 @@
+  pythonImportsCheck = [
+    "dsym"
+  ];
+
```
```suggestion

passthru.updateScript = nix-update-script { };

```

Commit:
adding dsym to realease notes as new package

Commit:
Changes as suggested in PR:

- formatted with nixfmt
- renaming package.nix
- remove comments
- remove duplicates
- apply new tree structure

nerdbude commented:
Thank you @fooker  and @acid-bong  for the suggestions. I learned a lot!

✅ All requested changes have been applied
✅ I've put the commits in the correct order

Commit:
resolve merge conflicts

acid-bong commented on nixos/modules/programs/dsym.nix:
```diff
@@ -0,0 +1,59 @@
+
+    dotfilePath = lib.mkOption {
+      type = lib.types.path;
+      default = "/home/${config.users.users.${config.users.defaultUser}.name}/.config/";
```
Hol up, `config.users.defaultUser` doesn't exist at all.

@nerdbude, admit: did some genAI (ChatGPT/Gemini/Grok/Mistral) write this?

acid-bong commented on pkgs/tools/misc/dsym/default.nix:
```diff
@@ -0,0 +1,35 @@
+    description = "Manage your dotfiles";
+    homepage = "https://github.com/nerdbude/dsym";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ 0x17 ];
```
Not necessary, they copied the recipe to `pkgs/by-name`, but forgot to remove from here

nerdbude commented:
All the merge conflicts should be fixed now. I removed the defaults.

acid-bong commented on nixos/doc/manual/release-notes/rl-2511.section.md:
```diff
@@ -50,7 +50,9 @@
-- [mautrix-discord](https://github.com/mautrix/discord), a Matrix-Discord puppeting/relay bridge. Available as [services.mautrix-discord](#opt-services.mautrix-discord.enable).
+[mautrix-discord](https://github.com/mautrix/discord), a Matrix-Discord puppeting/relay bridge. Available as [services.mautrix-discord](#opt-services.mautrix-discord.enable).
+
+[dsym](https://github.com/nerdbude/dsym), a small lightweight dotfile manager written in Python. Available as [programs.dsym](#opt-programs.dsym.enable).
```
Those lines must be list items, format them accordingly

acid-bong commented on nixos/modules/programs/dsym.nix:
```diff
@@ -0,0 +1,60 @@
+      dotfile_repo = cfg.dotfileRepo;
+      dotfile_path = toString cfg.dotfilePath;
+      dsym_path = toString cfg.dsymPath;
+    '';
```
```suggestion
    };
```
Next time check your files with a linter or set up an LSP in your editor

acid-bong reviewed (changes requested):
A couple more thangs. Also squash your edits, keep one commit per one thing (here it's new module (+ docs), new package and new maintainer entry)

acid-bong commented on nixos/modules/programs/dsym.nix:
```diff
@@ -0,0 +1,60 @@
+      dotfile_path = toString cfg.dotfilePath;
+      dsym_path = toString cfg.dsymPath;
+    '';
+  };
```
```suggestion
  };
  
  meta.maintainers = with lib.maintainers; [
    _0x17
  ];
```
Should also set yourself as a module maintainer

nerdbude committed:
Merge branch 'master' into dsym-addition

nerdbude commented:
ok. i think I solved all merge conflicts and all the checks run without any error.

Commit:
add package to release notes

fooker commented on nixos/modules/programs/dsym.nix:
```diff
@@ -0,0 +1,68 @@
+  };
+
+  meta.maintainers = with lib.maintainers; [
+
+    _0x17
+
```
Remove surrounding newlines

fooker commented on nixos/tests/dsym.nix:
```diff
@@ -0,0 +1,35 @@
+    nodes.machine =
+      { config, pkgs, ... }:
+      {
+        imports = [ "/home/null/GIT/nixpkgs/nixos/modules/programs/dsym.nix" ];
```
This seems strange. The module should be available without extra imports.

Commit:
adding suggestions

Commit:
removing result and edit .gitignore

yorenapu commented on pkgs/by-name/ds/dsym/package.nix:
```diff
@@ -0,0 +1,38 @@
+  src = fetchFromGitHub {
+    owner = "nerdbude";
+    repo = "dsym";
+    rev = version;
```
```suggestion
    tag = version;
```
use tag for a tag

Commit:
changes as reviewed
